### PR TITLE
Batching and Series naming from Select(ables)

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin ok with this?

File: odo/backends/sql.py [hunk 1]
```diff
@@ -96,6 +101,28 @@
...
+    batches of size `chunksize`.
+
+    Parameters
+    ----------
+    sel : sa.sql.Selectable
+        Selectable to execute
+    chunksize : int, optional, default 10000
+        Number of rows to fetch from the database
+    """
+    def rowterator(sel, chunksize=chunksize):
```
Comment (cpcloud) [inline odo/backends/sql.py]: @mrocklin is there a better way to do this?

File: odo/backends/sql.py [hunk 2]
```diff
@@ -293,39 +320,21 @@ def dshape_to_alchemy(dshape):
...
-    with engine.connect() as conn:
-        result = conn.execute(sel)
-        assert not dshape or isscalar(dshape)
-        result = list(result)[0][0]
-
-    return result
+    assert not dshape or isscalar(dshape), \
+        'dshape should be None or scalar, got %s' % dshape
+    with sel.bind.connect() as conn:
+        return conn.execute(sel).fetchall()[0][0]
```
Comment (cpcloud) [inline odo/backends/sql.py]: Whoops this shouldn't `fetchall`

File: odo/backends/sql.py (hunk 1 above)
Comment (mrocklin) [inline odo/backends/sql.py]: Probably, but I can't think of it.

**Comment by mrocklin** [comment]
Looks pretty good to me

**Comment by cpcloud** [comment]
cool thx for the review

**Comment by mrocklin** [comment]
Sorry my reviews have been sparse and shallow lately.

**Comment by cpcloud** [comment]
no problem! :) there's only so much time in the day

**Pull request merged**